src/alpaca_eval/decoders/openai.py

```diff
@@ -176,6 +177,9 @@ def _openai_completion_helper(
     if openai_api_keys is not None:
         openai.api_key = random.choice(openai_api_keys)
 
+    # set api base
+    openai.api_base = openai_api_base if openai_api_base is not None else 'https://api.openai.com/v1'
```

Comment:
Let’s avoid defaults that might change for OpenAI

Better:
```python
if openai_api_base is not None:
    openai_api_base = openai_api_base
```

Reply:
The openai api is reused in generating model answers and evaluators. I think we can set api_base each time to make sure it doesn't affect the subsequent evaluation process.

A potential solution could be:

```python
DEFAULT_OPENAI_API_BASE = openai.api_base

    # set api base
    openai.api_base = openai_api_base if openai_api_base is not None else DEFAULT_OPENAI_API_BASE
```

Another solution is to keep the default. `openai` package also hardcodes it [here](https://github.com/openai/openai-python/blob/041bf5a8ec54da19aad0169671793c2078bd6173/openai/__init__.py#L49)

What do you think?

Reply:
What was the issue with my if statement above? How is it different from setting DEFAULT_OPEN_API_BASE?

I’m also fine with your solution, I just want to avoid hardcoding variables which could change in future openai versions

Reply:
Thank you for suggesting the `if` statement solution. However, when `evaluate_from_model` is called, the `openai.api_base` will be changed to the model's endpoint. After that, if we were to call `evaluate` that uses GPT-4, the `openai_api_base` would be `None`, which would retain the changed endpoint instead of OpenAI's. 

Setting a default value like `DEFAULT_OPENAI_API_BASE = openai.api_base` and then using `openai_api_base = openai_api_base if openai_api_base is not None else DEFAULT_OPENAI_API_BASE` ensures that we don't accidentally use the changed endpoint and also avoids hardcoding variables which could change in future OpenAI versions. I hope that clarifies why I think setting a default value is a better approach.

Reply:
Thanks that makes sense! If you make that change, i’ll merge it.

Reply:
Thanks! I updated that and removed the hardcoded API endpoint.